iio/iio.c

```diff
@@ -1326,46 +1327,25 @@ static int iio_write_buffer(struct iiod_ctx *ctx, const char *device,
 	if (NO_OS_IS_ERR_VALUE(ret))
 		return ret;
 
-	available = dev->buffer.public.size - size;
-	bytes = no_os_min(available, bytes);
+	bytes = no_os_min(size, bytes);
```

Comment:
this should do the minimum between requested and available, not between requested and occupied